client-ts/signalr/src/HttpConnection.ts

```diff
@@ -116,25 +118,50 @@ export class HttpConnection implements IConnection {
-        if ((transport === null || transport === undefined) && availableTransports.length > 0) {
-            transport = TransportType[availableTransports[0]];
+    private createTransport(requestedTransport: TransportType | ITransport, availableEndpoints: IConnectionEndpoint[], requestedTransferFormat: TransferFormat): ITransport {
+        if (this.isITransport(requestedTransport)) {
+            this.logger.log(LogLevel.Trace, "Connection was provided an instance of ITransport, using that directly");
+            return requestedTransport;
+        }
+
+        for (const endpoint of availableEndpoints) {
+            const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
+            if (transport) {
+                return this.constructTransport(transport);
+            }
         }
-        if (transport === TransportType.WebSockets && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        throw new Error("Unable to initialize any of the available transports");
+    }
+
+    private constructTransport(transport: TransportType) {
+        if (transport === TransportType.WebSockets) {
             return new WebSocketTransport(this.options.accessTokenFactory, this.logger);
         }
-        if (transport === TransportType.ServerSentEvents && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        if (transport === TransportType.ServerSentEvents) {
             return new ServerSentEventsTransport(this.httpClient, this.options.accessTokenFactory, this.logger);
         }
-        if (transport === TransportType.LongPolling && availableTransports.indexOf(TransportType[transport]) >= 0) {
+
+        if (transport === TransportType.LongPolling) {
             return new LongPollingTransport(this.httpClient, this.options.accessTokenFactory, this.logger);
         }
+    }
 
-        if (this.isITransport(transport)) {
-            return transport;
+    private resolveTransport(endpoint: IConnectionEndpoint, requestedTransport: TransportType, requestedTransferFormat: TransferFormat): TransportType | null {
+        const transport = TransportType[endpoint.transport];
+        const transferFormats = endpoint.transferFormats.map((s) => TransferFormat[s]);
+        if (!requestedTransport || transport === requestedTransport) {
+            if (transferFormats.indexOf(requestedTransferFormat) >= 0) {
+                this.logger.log(LogLevel.Trace, `Selecting transport '${transport}'`);
+                return transport;
+            } else {
+                this.logger.log(LogLevel.Trace, `Skipping transport '${transport}' because it does not support the requested transfer format '${requestedTransferFormat}'`);
+            }
+        } else {
+            this.logger.log(LogLevel.Trace, `Skipping transport '${transport}' because it was disabled by the client.`);
```

Comment:
Fullstop on this log message but not others. Should pick a standard and stick to it.

Reply:
Looks like the standard is to have a `.`. Adding that.